spec/desks_spec.js

```diff
@@ -154,4 +158,66 @@ describe('desks', () => {
         // And clean up the desk we created
         desks.remove('Test Desk A');
     });
+
+    fit('can enforce incoming, outgoing and onstage rules', () => {
+        // Send stories go to incoming stage
+        desks.newDeskBtn.click();
+        desks.deskNameElement().sendKeys('Test Desk');
+        desks.deskSourceElement().sendKeys('Test Source A');
+        desks.setDeskType('authoring');
+        desks.actionSaveAndContinueOnGeneralTab();
+
+        desks.getNewStageButton().click();
+        desks.stageNameElement().sendKeys('Test Stage A');
+        desks.stageDescriptionElement().sendKeys('Test Desk A Stage A Description');
+        desks.setStageIncomingMacro('Validate for Publish');
+        desks.saveNewStage();
+
+        desks.getNewStageButton().click();
+        desks.stageNameElement().sendKeys('Test Stage B');
+        desks.stageDescriptionElement().sendKeys('Test Desk B Stage B Description');
+        desks.setStageOutgoingMacro('Validate for Publish');
+        desks.saveNewStage();
+
+        desks.getNewStageButton().click();
+        desks.stageNameElement().sendKeys('Test Stage C');
+        desks.stageDescriptionElement().sendKeys('Test Desk C Stage C Description');
+        desks.setStageMovedOntoMacro('Validate for Publish');
+        desks.saveNewStage();
+
+        desks.showTab('people');
+        desks.addUser('admin');
+        desks.close();
+
+        // confirm story is created on working stage
+        monitoring.openMonitoring();
+        browser.refresh();
+        workspace.selectDesk('Test Desk');
+        authoring.createTextItem();
+        authoring.writeTextToHeadline('new item');
+        authoring.save();
+        expect(monitoring.getGroupItems(0).count()).toBe(1);
+
+        // confirm incoming rule kicks in
+        authoring.sendTo('Test Desk', 'Test Stage A');
+        assertToastMsg('error', 'BODY_HTML is a required field');
```

Comment:
This is interrupt the test suite, instead of run all tests
```
xvfb-run -a -s '-ac -screen 0 1920x1080x24' protractor-flake --max-attempts=2 -- protractor.conf.js --stackTrace --verbose --troubleshoot --baseUrl http://localhost --params.baseBackendUrl http://localhost/api --specs spec/archived_spec.js,spec/authoring_spec.js,spec/content_profile_spec.js,spec/content_spec.js,spec/dashboard_spec.js,spec/desks_management_spec.js,spec/desks_spec.js,spec/dictionaries_spec.js,spec/fetch_spec.js,spec/highlights_spec.js,spec/ingest_provider_spec.js,spec/ingest_settings_spec.js,spec/legal_archive_spec.js,spec/login_spec.js,spec/marked_desks_spec.js
```
it only runs one:
```
.


Ran 1 of 88 specs
1 spec, 0 failures
Finished in 58.679 seconds
```
It's from log: https://test.superdesk.org/logs/3/dev/latest/sdcpr-1632/check-e2e--part1.log.htm